internal/dag/builder.go

```diff
@@ -478,6 +478,12 @@ func (b *Builder) computeHTTPProxy(proxy *projcontour.HTTPProxy) {
 		// tls is valid if passthrough == true XOR secretName != ""
 		tlsValid = tls.Passthrough != !isBlank(tls.SecretName)
 
+		// passthrough cannot be combined with secretName or clientValidation
+		if tls.Passthrough && (!isBlank(tls.SecretName) || (tls.ClientValidation != nil)) {
```

Comment:
I think that it is enough here to check the client validation. The secret name is check for TCP proxy, though that check could arguably be tighter.

```Go
if tls.Passthrough && tls.ClientValidation != nil {
        sw.SetInvalid("Spec.VirtualHost.TLS ...")
        return
}
```

Reply:
The code around this had changed a bit so I adapted this suggestion a bit, but implemented the idea.